### PR TITLE
Create task on db

### PR DESCRIPTION
**Pull request by akib-raihan-sazim**
**Description of changes**
- Integrated the createTask endpoint in the apiSlice for adding new task.
- Replaced the old addTask function with new addTask function, now the task will be save on db.

**Associated ticket link**
https://trello.com/c/x5O3OvHy

**Screenshot / Screen Recording**

https://github.com/user-attachments/assets/27a67420-58f0-408c-9ac5-5aeb0ddb57fa

### CODE REVIEW

File: modules/HomePage/components/TaskContext/TaskContext.tsx [hunk 1]
```diff
@@ -87,6 +87,8 @@ export const TasksProvider = ({ children }: { children: ReactNode }) => {
 
   const {data: fetchedTasks} = useGetTasksQuery();
 
+  const [createTask] = useCreateTaskMutation();
```
Comment (ishmam-iqbal-sazim) [inline modules/HomePage/components/TaskContext/TaskContext.tsx:90]: The context should be used for state management, not for api calls. Please create a Tech Debt ticket to move all the api calls and hooks to their relevant components.

File: shared/redux/rtk-apis/apiSlice.ts [hunk 2]
```diff
@@ -8,8 +8,16 @@ export const apiSlice = createApi({
     endpoints: (builder) => ({
         getTasks: builder.query<ITask[], void>({
             query: () => '/tasks',
-        })
+        }),
+
+        createTask: builder.mutation<ITask, Partial<ITask>>({
```
Comment (ishmam-iqbal-sazim) [inline shared/redux/rtk-apis/apiSlice.ts:13]:
```suggestion
        createTask: builder.mutation<ITask, TCreateTaskDto>({
```
Might have to create a new type. Best not to bypass type specificity by usage of Partials.

File: modules/HomePage/components/TaskContext/TaskContext.tsx [hunk 3]
```diff
@@ -113,10 +115,15 @@ export const TasksProvider = ({ children }: { children: ReactNode }) => {
...
-    saveHistoryState();
+  const addTask = async (task: Omit<ITask, "id" | "completed">) => {
+    try {
+      const result = await createTask(task).unwrap();
+      const newTask = { ...result, completed: false };
+      setTasks((prevTasks) => sortTasks([...prevTasks, newTask]));
+      saveHistoryState();
+    } catch (error) {
+      console.error("Failed to create task:", error);
+    }
```
Comment (ishmam-iqbal-sazim) [inline modules/HomePage/components/TaskContext/TaskContext.tsx:126]: Same comment as above. No changes necesary in this PR.

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Approved pending comments.

**Review by ragib-hossain-sazim — APPROVED** [review]

**Pull request merged**